server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -352,10 +352,17 @@ protected void startProbe(TransportAddress transportAddress) {
         }
     }
 
+    public List<DiscoveryNode> getMastersOfPeers() {
+        synchronized (mutex) {
+            return peersByAddress.values().stream().flatMap(p -> p.lastKnownMasterNode.stream()).distinct().toList();
+        }
+    }
```

Comment:
I believe this is used to populate the new `List<DiscoveryNode> mastersOfPeers`. I was originally assuming we report master seen by each node to verify each have the same, however here it is using `distinct()`. Why not using a set to make it obvious there are no duplicates?

Reply:
Fair point, done in 8802ebf